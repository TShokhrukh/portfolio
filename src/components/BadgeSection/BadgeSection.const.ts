import { cn } from '@bem-react/classname'

export const cnBadgeSection = cn('BadgeSection')
