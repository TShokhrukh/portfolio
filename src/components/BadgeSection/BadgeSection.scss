.BadgeSection-Head {
  text-align: center;
}

.BadgeSection-Title {
  margin-top: $spacer * 3;
}

.BadgeSection-Body {
  margin-top: $spacer * 2;
}
