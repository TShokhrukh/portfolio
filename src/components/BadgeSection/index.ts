export * from './build'
export * from './__types__'
export { BadgeSection as BadgeSectionBase } from './BadgeSection'
