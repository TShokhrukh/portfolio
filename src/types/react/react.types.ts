import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
