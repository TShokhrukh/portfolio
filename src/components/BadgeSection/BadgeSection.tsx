import React, { FC } from 'react'
import { Typography } from '~/components/Typography/build'
import { Badge } from '~/components/Badge/build'
import { cnBadgeSection } from './BadgeSection.const'
import type { IBadgeSectionProps } from './__types__'
import './BadgeSection.scss'

export const BadgeSection: FC<IBadgeSectionProps> = ({
  badge,
  title,
  children,
  className,
  component: Component = 'div',
  ...props
}) => {
  return (
    <Component {...props} className={cnBadgeSection({}, [className])}>
      <div className={cnBadgeSection('Head', {})}>
        <Badge color='purple' className={cnBadgeSection('Badge', {})}>
          {badge}
        </Badge>
        <Typography variant='h3' color='white' className={cnBadgeSection('Title', {})}>
          {title}
        </Typography>
      </div>
      <div className={cnBadgeSection('Body', {})}>
        {children}
      </div>
    </Component>
  )
}
