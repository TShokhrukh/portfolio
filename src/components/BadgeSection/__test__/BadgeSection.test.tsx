/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { BadgeSection } from '../BadgeSection'

describe('BadgeSection', () => {
  test('must return the component template', () => {
    render(<BadgeSection data-testid='badge_section' badge={<span>badge</span>} title={'Title'}>content</BadgeSection>)

    expect(screen.getByTestId('badge_section')).toMatchSnapshot()
  })

  test('must change tag name by component prop', () => {
    render(<BadgeSection component='section' data-testid='badge_section' />)

    expect(screen.getByTestId('badge_section').tagName).toBe('SECTION')
  })
})
