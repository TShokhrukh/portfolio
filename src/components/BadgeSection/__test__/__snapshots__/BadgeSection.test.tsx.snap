// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeSection must return the component template 1`] = `
<div
  class="BadgeSection"
  data-testid="badge_section"
>
  <div
    class="BadgeSection-Head"
  >
    <span
      class="Badge Badge_color_purple BadgeSection-Badge"
    >
      <span>
        badge
      </span>
    </span>
    <h3
      class="Typography Typography_color_white Typography_variant_h3 BadgeSection-Title"
    >
      Title
    </h3>
  </div>
  <div
    class="BadgeSection-Body"
  >
    content
  </div>
</div>
`;
